src/services/patient/patient.service.js

```diff
@@ -569,31 +560,31 @@ module.exports.update = (args, { req }) =>
       let doc = Object.assign(cleaned, { _id: id });
 
       // Insert/update our patient record
-      collection.findOneAndUpdate({ id: id }, { $set: doc }, { upsert: true }, (err2, res) => {
-        if (err2) {
-          logger.error('Error with Patient.update: ', err2);
-          return reject(err2);
-        }
-
+      collection.findOneAndUpdate({ id: id }, { $set: doc }, { upsert: true }).then(res => {
         // save to history
         let history_collection = db.collection(`${COLLECTION.PATIENT}_${base_version}_History`);
 
         let history_patient = Object.assign(cleaned, { id: id });
 
         // Insert our patient record to history but don't assign _id
-        return history_collection.insertOne(history_patient, (err3) => {
-          if (err3) {
+        return history_collection.updateOne({ id: id }, {$set: history_patient}, { upsert: true })
+          .then( () => {
+            return resolve({
+              id: id,
+              created: res && res.lastErrorObject ? !res.lastErrorObject.updatedExisting : false,
```

Comment:
this was updated to support null check and default to false.  This was a bug previously where the first put would fail